site/src/app/[domain]/[language]/[[...path]]/page.tsx

```diff
@@ -1,33 +1,71 @@
 import { gql, previewParams } from "@comet/cms-site";
+import { ExternalLinkBlockData, InternalLinkBlockData, RedirectsLinkBlockData } from "@src/blocks.generated";
 import { documentTypes } from "@src/documents";
-import { GQLPageTreeNodeScopeInput } from "@src/graphql.generated";
+import { GQLPageTreeNodeScope } from "@src/graphql.generated";
 import { createGraphQLFetch } from "@src/util/graphQLClient";
-import { notFound } from "next/navigation";
+import { getSiteConfigForDomain } from "@src/util/siteConfig";
+import { notFound, redirect } from "next/navigation";
 
 import { GQLDocumentTypeQuery, GQLDocumentTypeQueryVariables } from "./page.generated";
 
 const documentTypeQuery = gql`
-    query DocumentType($path: String!, $scope: PageTreeNodeScopeInput!) {
-        pageTreeNodeByPath(path: $path, scope: $scope) {
+    query DocumentType(
+        $skipPage: Boolean!
+        $path: String!
+        $scope: PageTreeNodeScopeInput!
+        $redirectSource: String!
+        $redirectScope: RedirectScopeInput!
+    ) {
+        pageTreeNodeByPath(path: $path, scope: $scope) @skip(if: $skipPage) {
             id
             documentType
         }
+        redirectBySource(source: $redirectSource, sourceType: path, scope: $redirectScope) {
+            target
+        }
     }
 `;
 
-export default async function Page({ params: { path, domain, language } }: { params: { path: string[]; domain: string; language: string } }) {
-    const scope = { domain, language };
-
+export default async function Page({ params }: { params: { path: string[]; domain: string; language: string } }) {
     const { previewData } = (await previewParams()) || { previewData: undefined };
     const graphqlFetch = createGraphQLFetch(previewData);
+    const siteConfig = getSiteConfigForDomain(params.domain);
+
+    const skipPage = !siteConfig.scope.languages.includes(params.language);
```

Comment:
Should we add a comment explaining this? Or do you think it's obvious enough?

Reply:
yes, this is not very obvious

maybe something like that?
```suggestion
    const skipPage = !siteConfig.scope.languages.includes(params.language); // redirects are not language scoped, language is invalid we still need to load them (and skip pages)
```

Reply:
```suggestion
    // Redirects are scoped by domain only, not by language.
    // If the language param isn't a valid language, it may still be the first segment of a redirect source.
    // In that case we skip resolving page and only check if the path is a redirect source.
    const skipPage = !siteConfig.scope.languages.includes(params.language);
```